files-override/ts/app/app.ts

```diff
@@ -0,0 +1,25 @@
+import Application from '@ember/application';
+// This can be removed if you have no need for compatModules
+// @ts-expect-error TODO: add types to @embroider/core/entrypoint
+import compatModules from '@embroider/core/entrypoint';
+import Resolver from 'ember-resolver';
+import loadInitializers from 'ember-load-initializers';
+import config from './config/environment';
+
+// This can be removed if you have no need for compatModules
+// @ts-expect-error window does not have define 
+const d = window.define;
```

Comment:
if Ed finishes his work on passing config modules to loadInitializers we won't need this whole block in this file and yet another place where the file doesn't need to differ between JS and TS

Reply:
YAAAS